### PR TITLE
refactor(flow): Port wallet edge cases to wallet menu flow

### PR DESCRIPTION
**Pull request by amanCypherock**

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4970 Wallet edge cases redirection](https://app.clickup.com/t/9002019994/PRF-4970)

File: src/menu/main_menu.c [hunk 1]
```diff
@@ -184,18 +187,15 @@ static void ignore_p0_handler(engine_ctx_t *ctx,
   ignore_p0_event();
 }
 
-static void get_wallet_id_from_list_selection(uint8_t selection,
-                                              uint8_t *wallet_id) {
-  ASSERT(0 != selection && 4 >= selection);
+static void get_flash_wallet_from_list_selection(uint8_t selection,
+                                                 Flash_Wallet **flash_wallet) {
+  ASSERT(0 != selection && 4 >= selection && NULL != flash_wallet);
```
Comment (kartikCypherock) [inline src/menu/main_menu.c:192]: Check the range for selection. I think it can be 4 (1 indexed)

File: src/menu/wallet_menu.c [hunk 2]
```diff
@@ -120,23 +122,24 @@ static void wallet_menu_handler(engine_ctx_t *ctx,
...
+ * object of selected wallet
  */
 static void ignore_p0_handler(engine_ctx_t *ctx,
                               p0_evt_t p0_evt,
                               const void *data_ptr);
 /*****************************************************************************
  * STATIC VARIABLES
  *****************************************************************************/
-static uint8_t wallet_id[WALLET_ID_SIZE] = {0};
+static Flash_Wallet flash_wallet = {0};
```
Comment (kartikCypherock) [inline src/menu/wallet_menu.c:134]: Can you check if there is already a global variable of type `Flash_Wallet` present in the code? Aim should be to reduce duplicity..

Also, do you think we require the complete structure? Like pow_challenge` data? Or just the states/ wallet id/ name

Comment (amanCypherock) [reply src/menu/wallet_menu.c:134]: Picked the `Flash_Wallet` structure as it contains all important state variables. The function `get_flash_wallet_from_list_selection` returns the pointer to global variable of that Flash_Wallet object in flash_ram_instance.

We don't require the complete structure, the share and challenge are unnecessary for this purpose.

File: src/menu/main_menu.c (hunk 1 above)
Comment (amanCypherock) [reply src/menu/main_menu.c:192]: No change has been made to the range of selection. The assert checks for 1 to 4 inclusive.

File: src/menu/wallet_menu.c (hunk 2 above)
Comment (kartikCypherock) [reply src/menu/wallet_menu.c:134]: In that case saving a copy of Flash_Wallet seems duplicate.. as we can fetch from existing functions everytime we require the info.. not making this comment blocking though..

**Review by kartikCypherock — APPROVED** [review]

File: src/menu/wallet_menu.c (hunk 2 above)
Comment (amanCypherock) [reply src/menu/wallet_menu.c:134]: In that case, would it work to just keep the flash_wallet object pointer for that wallet?

Comment (amanCypherock) [reply src/menu/wallet_menu.c]: updated

**Comment by kartikCypherock** [comment]
LGTM

**Pull request merged**